### PR TITLE
range_plus_one suggestion should not remove braces fix

### PR DESCRIPTION
**Pull request by daubaris**
Looking into #3103 decided to go with the first suggestion. 

https://github.com/rust-lang-nursery/rust-clippy/issues/3103#issuecomment-417368043

Added information to the known problems that lint adds unnecessary parentheses if the expression is not wrapped in a pair but starts with a opening parenthesis and ends with a closing one.

### CODE REVIEW

File: clippy_lints/src/ranges.rs [hunk 1]
```diff
@@ -49,7 +49,10 @@ declare_clippy_lint! {
 /// **Why is this bad?** The code is more readable with an inclusive range
 /// like `x..=y`.
 ///
-/// **Known problems:** None.
+/// **Known problems:** Will add unnecessary pair of parentheses when the
+/// expression is not wrapped in a pair but starts with a opening parenthesis
+/// and ends with a closing one.
+/// I.e: let _ = (f()+1)..(f()+1) results in let _ = ((f()+1)..(f()+1)).
```
Comment (flip1995) [inline clippy_lints/src/ranges.rs]: Backticks "\`" around code missing. "results in `let _ = ((f()+1)..=f())`"

File: tests/ui/range_plus_minus_one.stderr [hunk 2]
```diff
@@ -42,7 +42,7 @@ error: an inclusive range would be more readable
   --> $DIR/range_plus_minus_one.rs:28:13
    |
 28 |     let _ = (f()+1)..(f()+1);
-   |             ^^^^^^^^^^^^^^^^ help: use: `(f()+1)..=f()`
+   |             ^^^^^^^^^^^^^^^^ help: use: `((f()+1)..=f())`
```
Comment (flip1995) [inline tests/ui/range_plus_minus_one.stderr]: Please add another test especially for this issue.

**Comment by daubaris** [comment]
Hello, I cannot test my work because of the errors that I'm getting when I run `cargo +master test`:
`reg.register_lint_group` - expects two parameters and `Could not find Guard in hir`.

File: clippy_lints/src/ranges.rs [hunk 3]
```diff
@@ -49,7 +49,10 @@ declare_clippy_lint! {
 /// **Why is this bad?** The code is more readable with an inclusive range
 /// like `x..=y`.
 ///
-/// **Known problems:** None.
+/// **Known problems:** Will add unnecessary pair of parentheses when the
+/// expression is not wrapped in a pair but starts with a opening parenthesis
+/// and ends with a closing one.
+/// I.e: `let _ = (f()+1)..(f()+1)` results in `let _ = ((f()+1)..(f()+1))`.
```
Comment (flip1995) [inline clippy_lints/src/ranges.rs]: The second `let _ = ((f()+1)..(f()+1))` should be `let _ = ((f()+1)..=f())`

**Comment by Manishearth** [comment]
Hi! Could you leave the requested comment in https://github.com/rust-lang-nursery/rust-clippy/issues/3230 for relicensing?

**Pull request merged**